rfc/helm-import.md

```diff
@@ -0,0 +1,515 @@
+
+```typescript
+  public get moduleNames() {
+    return [this.chartName];
+  }
+
+  protected async generateTypeScript(code: CodeMaker) {
+
+    emitHelmHeader(code);
+
+    let schema: JSONSchema4 | undefined;
+
+    if (this.chartSchemaPath !== undefined) {
+      JSON.parse(fs.readFileSync(this.chartSchemaPath, 'utf-8'));
+    } else {
+      schema = undefined;
+    }
+
+    const types = new TypeGenerator({
+      definitions: schema?.definitions,
+    });
+
+    generateHelmConstruct(types, {
+      schema: schema,
+      chartName: this.chartName,
+      chartUrl: this.chartUrl,
+      chartVersion: this.chartVersion,
+      fqn: this.chartName,
+    });
+
+    code.line(types.render());
+
+    cleanup(this.tmpDir);
+  }
+```
+
+For code generation, we already utilize [codemaker](https://www.npmjs.com/package/codemaker). The `emitHelmHeader(code)`   function would be responsible for adding any relevant imports for the generated construct.
+
+```typescript
+/**
+ * Emit imports for generated helm construct
+ * @param code CodeMaker instance
+ */
+export function emitHelmHeader(code: CodeMaker) {
+  code.line('// generated by cdk8s');
+  code.line('import { Helm } from \'cdk8s\';');
+  code.line('import { Construct } from \'constructs\';');
+  code.line();
+}
+```
+
+If we did find `values.schema.json` in the helm chart, the definitions of the schema as passed in to the [TypeGenerator](https://github.com/cdklabs/json2jsii). TypeGenerator makes it possible to emit structs for a schema and also enables to emit custom types.
+
+We then invoke `generateHelmConstruct` function. In this function, we are emitting: 
+  * A custom type. This would be our helm construct.
+    * It's constructor/initializer.
+  * A `values` property `struct` if schema is defined.
+  * An interface representing properties of the construct.
+
+```typescript
+export function generateHelmConstruct(typegen: TypeGenerator, def: HelmObjectDefinition) {
+
+  const chartName = TypeGenerator.normalizeTypeName(def.chartName);
+  const schema = def.schema;
+  const repoUrl = def.chartUrl;
+  const chartVersion = def.chartVersion;
+
+  typegen.emitCustomType(chartName, code => {                   // <--------------- Creating custom type
+
+    const valuesInterface = `${chartName}ValuesProps`;          // <--------------- Creating 'values' struct
+    if (schema !== undefined) {
+      typegen.emitType(valuesInterface, schema, def.fqn);
+    }
+
+    emitPropsInterface();                                       // <--------------- Creating construct properties
+
+    code.line();
+
+    emitConstruct();                                            // <--------------- Creating construct for helm chart
+
+    function emitPropsInterface() {
+      code.openBlock(`export interface ${chartName}Props`);
+
+      code.line('readonly chart?: string;');
+      code.line('readonly repo?: string;');
+      code.line('readonly version?: string;');
```

Comment:
These shouldn't be part of the interface though. We should just set them to the value (that we already know) and use them in the implementation.

Reply:
Ahh, correct. Will change it.